### PR TITLE
Feat: added support for unused DaemonSets

### PR DESCRIPTION
**Pull request by doronkg**
This PR adds a new functionality to `kor` by finding unused DaemonSets.
I've considered DaemonSets to be unused, if the following condition is met:
```
status:
  currentNumberScheduled: 0
```

Meaning the number of nodes that should be running at least 1 daemon pod and are supposed to run the daemon pod, is equal to zero.
More information about DaemonSets' status fields could be found in these [docs](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.28/#daemonsetstatus-v1-apps).

### CODE REVIEW

**Comment by doronkg** [comment]
Conflicts might raise in files `pkg/kor/delete.go` (#201) & `charts/kor/templates/role.yaml` (#202).
Should be taken into account when merging.

File: pkg/kor/create_test_resources.go [hunk 1]
```diff
@@ -321,4 +321,33 @@ func CreateTestReplicaSet(namespace, name string, specReplicas *int32, status *a
...
+}
+
+func CreateTestDaemonSet(namespace, name string, labels map[string]string, status *appsv1.DaemonSetStatus) *appsv1.DaemonSet {
+	return &appsv1.DaemonSet{
+		ObjectMeta: v1.ObjectMeta{
+			Namespace: namespace,
+			Name:      name,
+			Labels:    labels,
+		},
+		Spec: appsv1.DaemonSetSpec{
```
Comment (yonahd) [inline pkg/kor/create_test_resources.go:333]: nit: do we need the spec part of the daemonset?

**Review by yonahd — COMMENTED** [review]
Lot great overall small comment

File: pkg/kor/create_test_resources.go (hunk 1 above)
Comment (doronkg) [reply pkg/kor/create_test_resources.go]: Was indeed excessive, test ran successfully without it. Removed.
Thanks

**Review by yonahd — APPROVED** [review]
LGTM
Thanks

**Pull request merged**